diff --git a/mbedtls-4.0.0/library/CMakeLists.txt b/mbedtls-4.0.0/library/CMakeLists.txt
--- a/mbedtls-4.0.0/library/CMakeLists.txt
+++ b/mbedtls-4.0.0/library/CMakeLists.txt
@@ -147,11 +147,11 @@ if(CMAKE_COMPILER_IS_MSVC)
     endif()
 endif()
 
-if(CMAKE_C_COMPILER_ID MATCHES "AppleClang")
+if(CMAKE_C_COMPILER_ID MATCHES "AppleClang" AND NOT MBEDTLS_PRESERVE_TOOLCHAIN_ARCHIVER)
     set(CMAKE_C_ARCHIVE_CREATE   "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
     set(CMAKE_C_ARCHIVE_FINISH   "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
 endif()
-if(CMAKE_CXX_COMPILER_ID MATCHES "AppleClang")
+if(CMAKE_CXX_COMPILER_ID MATCHES "AppleClang" AND NOT MBEDTLS_PRESERVE_TOOLCHAIN_ARCHIVER)
     set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
     set(CMAKE_CXX_ARCHIVE_FINISH "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
 endif()
diff --git a/mbedtls-4.0.0/tf-psa-crypto/core/CMakeLists.txt b/mbedtls-4.0.0/tf-psa-crypto/core/CMakeLists.txt
--- a/mbedtls-4.0.0/tf-psa-crypto/core/CMakeLists.txt
+++ b/mbedtls-4.0.0/tf-psa-crypto/core/CMakeLists.txt
@@ -82,11 +82,11 @@ if(CMAKE_COMPILER_IS_MSVC)
     endif()
 endif()
 
-if(CMAKE_C_COMPILER_ID MATCHES "AppleClang")
+if(CMAKE_C_COMPILER_ID MATCHES "AppleClang" AND NOT MBEDTLS_PRESERVE_TOOLCHAIN_ARCHIVER)
     set(CMAKE_C_ARCHIVE_CREATE   "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
     set(CMAKE_C_ARCHIVE_FINISH   "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
 endif()
-if(CMAKE_CXX_COMPILER_ID MATCHES "AppleClang")
+if(CMAKE_CXX_COMPILER_ID MATCHES "AppleClang" AND NOT MBEDTLS_PRESERVE_TOOLCHAIN_ARCHIVER)
     set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
     set(CMAKE_CXX_ARCHIVE_FINISH "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
 endif()
diff --git a/mbedtls-4.0.0/tf-psa-crypto/drivers/builtin/CMakeLists.txt b/mbedtls-4.0.0/tf-psa-crypto/drivers/builtin/CMakeLists.txt
--- a/mbedtls-4.0.0/tf-psa-crypto/drivers/builtin/CMakeLists.txt
+++ b/mbedtls-4.0.0/tf-psa-crypto/drivers/builtin/CMakeLists.txt
@@ -22,11 +22,11 @@ if(CMAKE_COMPILER_IS_MSVC)
     endif()
 endif()
 
-if(CMAKE_C_COMPILER_ID MATCHES "AppleClang")
+if(CMAKE_C_COMPILER_ID MATCHES "AppleClang" AND NOT MBEDTLS_PRESERVE_TOOLCHAIN_ARCHIVER)
     set(CMAKE_C_ARCHIVE_CREATE   "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
     set(CMAKE_C_ARCHIVE_FINISH   "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
 endif()
-if(CMAKE_CXX_COMPILER_ID MATCHES "AppleClang")
+if(CMAKE_CXX_COMPILER_ID MATCHES "AppleClang" AND NOT MBEDTLS_PRESERVE_TOOLCHAIN_ARCHIVER)
     set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
     set(CMAKE_CXX_ARCHIVE_FINISH "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
 endif()
